### PR TITLE
Add `firstindex` method and tests

### LINKED ISSUE #99: `firstindex` method error in color scheme
Hi, I am not sure if this is expected behavior, but when one attempts to access the first color in a color scheme using `begin` instead of 1, a method error happens. Here is a MWE:
```julia
ColorSchemes.Dark2_3[begin] # error
ColorSchemes.Dark2_3[1] # okay
ColorSchemes.Dark2_3[end] # this works though
```
```julia
julia> cs = ColorSchemes.Dark2_3[begin]
ERROR: MethodError: no method matching firstindex(::ColorScheme{Vector{ColorTypes.RGB{Float64}}, String, String})
Closest candidates are:
  firstindex(::Any, ::Any) at abstractarray.jl:402
  firstindex(::Union{Tables.AbstractColumns, Tables.AbstractRow}) at C:\Users\stelmo\.julia\packages\Tables\PxO1m\src\Tables.jl:177
  firstindex(::Interpolations.KnotIterator) at C:\Users\stelmo\.julia\packages\Interpolations\jPlKV\src\iterate.jl:156
  ...
Stacktrace:
 [1] top-level scope
   @ REPL[206]:1
```

### PR DESCRIPTION
**Pull request by stelmo**
Fixes #99 when merged. Adds `firstindex` method and adds some indexing tests.

### CODE REVIEW

**Comment by stelmo** [comment]
Not sure why 1.3 failed :/

**Comment by cormullion** [comment]
> Not sure why 1.3 failed :/

`firstindex()` was introduced in Julia version v1.5 I think... 😀

**Comment by rafaqz** [comment]
Yeah `begin` didn't exist in 1.3. Might need a conditional to test it.

File: test/runtests.jl [hunk 1]
```diff
@@ -43,6 +43,13 @@ monalisa = ColorScheme([
     @inferred get(monalisa, 0.0)
     @test get(monalisa, 0.0) != get(monalisa, 0.5)
     @test monalisa[end] ≈ monalisa[1.0]
+    
+    # indexing tests
+    if Base.VERSION > v"1.6.0" # begin was only introduce later
+        @test monalisa[begin] == first(monalisa)
+        @test monalisa[1] == first(monalisa)
+        @test monalisa[end] == last(monalisa)
+    end
```
Comment (stelmo) [inline test/runtests.jl:52]: It still breaks here even after I put the condition testing on...

**Comment by stelmo** [comment]
Hmm, I added a conditional in the test (https://github.com/JuliaGraphics/ColorSchemes.jl/pull/100/files#r969470216) but it still doesn't pass 😕

**Comment by stelmo** [comment]
Hmm it seems to be a load error actually `ERROR: LoadError: syntax: unexpected "]"`. Can I make a PR to drop support for testing against Julia 1.3? Or is there a special reason for using that version?

**Comment by cormullion** [comment]
I wonder if it could be done with `eval`?

```julia
if VERSION > v"1.5.0"
    eval((:Base.firstindex() = ...
end
```

Yes, specifying Julia as v1.6.0 (LTS) is probably a good idea, although there are a lot of packages that depend on ColorSchemes, so it should be considered in context.

**Comment by cormullion** [comment]
That's a pity... :(

**Comment by stelmo** [comment]
Yep, it didn't work :( Hmm I can remove the tests from the PR or I can make a PR that bumps up the testing requirements. However, it seems that the package might cause problems if anyone is still using Julia 1.3... Either way is fine with me though!

**Comment by cormullion** [comment]
To be honest, I'm tempted to just not test the `begin` syntax until we drop 1.3 support later, when we can add this test back in  again... :) Prefer to let depending packages (Makie, Plots, etc) make the move first... :)

**Comment by stelmo** [comment]
No problem! I will put a to do note in the test and comment it out : )

**Pull request merged**